service/controller/app/v1/key/key.go

```diff
@@ -65,6 +65,18 @@ func InCluster(customResource v1alpha1.App) bool {
 	return customResource.Spec.KubeConfig.InCluster
 }
 
+func KubeConfigFinalizer(customResource v1alpha1.App) string {
+	return fmt.Sprintf("%s/%s", label.FinalizerPrefix, customResource.GetName())
```

Comment:
I thought about something like this. And then drop the changes in the label package. 

```suggestion
	return fmt.Sprintf("app-operator.giantswarm.io/%s", customResource.GetName())
```